modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/ScriptMediator.java

```diff
@@ -403,8 +403,12 @@ private void processJSONPayload(MessageContext synCtx, ScriptMessageContext scri
         if (JsonUtil.hasAJsonPayload(messageContext)) {
             try {
                 String jsonPayload = JsonUtil.jsonPayloadToString(messageContext);
-                String scriptWithJsonParser = "JSON.parse(JSON.stringify(" + jsonPayload + "))";
-                jsonObject = this.jsEngine.eval('(' + scriptWithJsonParser + ')');
+                if (language.equals(NASHORN_JAVA_SCRIPT)) {
```

Comment:
Better to have
`NASHORN_JAVA_SCRIPT.equals(language)`

Reply:
Done